pkg/controller/statusmanager/pod_watcher.go

```diff
@@ -75,11 +76,11 @@ func (s *StatusManager) initInformersFor(clusterName, namespace string, stateful
 
 // AddPodWatcher wires up the PodWatcher to the controller-manager.
 func (s *StatusManager) AddPodWatcher(mgr manager.Manager) error {
-	s.initInformersFor("", metav1.NamespaceAll, false)
```

Comment:
in the hypershift hosted control plane case, why would we watch all namespaces ?

Reply:
We want to watch all of the resources created by the CNO in the hosted cluster.

Reply:
but that would not be on the same API endpoint, would it? This question is probably out of scope of this PR, sorry. Happy to take it elsewhere.

Reply:
Correct, CNO talks to both the management and hosted clusters API servers. 
`initInformersFor` uses the `clusterName` to determine which client to use (empty string means hosted cluster API server)